### PR TITLE
#157244155 Integrate paystack payment

### PR DESCRIPTION
**Pull request by Felglitzzz**
#### What does this PR do?
Integrates paystack payment as one of the payment options for users

#### Description of Task to be completed?

- create a custom plugin for paystack payment option
- Tweak the checkout files to reflect needful fields
- Integrate the paystack inline API
- Integrate server methods to allow for payments via paystack

#### How should this be manually tested?

- npm run git clone `https://github.com/andela/nazgul-rc.git` to clone the repository
- navigate to the nazgul-rc folder
- run `curl https://install.meteor.com/ | sh`
- run `npm install -g reaction-cli`
- install dependencies with`meteor npm install`
- run `reaction` to start up server
- once app is running, navigate to the app using localhost:3000
- navigate to shop page
- add items to checkout by clicking the cart icon
- enter the needful details, select a shipping rate and fill your details in the form
- click on "Complete your order" to make payment
- follow the payment instructions

#### What are the relevant pivotal tracker stories?
#157245700
#### Screenshots (if appropriate)
paystack
paystack2
paystack3
paystack4

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -27,7 +29,6 @@
         "destructuring": true,
         "forOf": true,
         "generators": false,
-        "modules": true,
```
Comment (morekoya) [inline .eslintrc:30]: Are you sure the removal of this isn't what is causing your import parsing error?

File: .bashrc [hunk 2]
```diff
@@ -0,0 +1 @@
+ulimit -n 65536 65536
```
Comment (morekoya) [inline .bashrc]: I'm pretty sure your `.bashrc` file should not have been pushed

File: imports/plugins/custom/payments-paystack/client/checkout/paystack.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+        const cart = Cart.findOne().getTotal();
+        const amount = Math.round(cart * 100);
+        const template = this.template;
+
+        const formData = {
+          key: publicKey,
+          name: doc.payerName,
+          email: doc.payerEmail,
+          reference: Random.id(),
```
Comment (morekoya) [inline imports/plugins/custom/payments-paystack/client/checkout/paystack.js:63]: Isn't this supposed to be someway of tracking the product purchased? Won't it be better for the reference to be the product id?

File: imports/plugins/custom/payments-paystack/client/settings/components/paystackSettingsForm.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+          />
+
+          <label className="control-label">
+            <Translation defaultValue="Payment provider supported methods" i18nKey="reaction-payments.paymentSettings.supportedMethodsLabel"/>
+          </label>
+          <br/>
+
+          <div>
+            <Checkbox
+              label="Authorize"
```
Comment (morekoya) [inline imports/plugins/custom/payments-paystack/client/settings/components/paystackSettingsForm.js]: I think that by the time a string is being repeated more three times, you might want to save it as a constant. This makes it easier to make make file wide changes when necessary.

File: imports/plugins/custom/payments-paystack/server/i18n/en.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
+[{
+  "language": "English",
+  "i18n": "en",
+  "ns": "paystack-paymentmethod",
+  "translation": {
+    "reaction-payments": {
+      "admin": {
+        "shortcut": {
+          "paystackPaymentProviderLabel": "paystack Payment",
```
Comment (morekoya) [inline imports/plugins/custom/payments-paystack/server/i18n/en.json]: Any particular reason why your phrases start with small letters?

File: .bashrc (hunk 2 above)
Comment (Felglitzzz) [reply .bashrc]: okay, I will fix it on my next push

File: imports/plugins/custom/payments-paystack/client/settings/components/paystackSettingsForm.js (hunk 4 above)
Comment (Felglitzzz) [reply imports/plugins/custom/payments-paystack/client/settings/components/paystackSettingsForm.js]: okay, noted

File: imports/plugins/custom/payments-paystack/server/i18n/en.json (hunk 5 above)
Comment (Felglitzzz) [reply imports/plugins/custom/payments-paystack/server/i18n/en.json]: No reason, an oversight on my part

File: imports/plugins/custom/payments-paystack/client/checkout/paystack.js (hunk 3 above)
Comment (Felglitzzz) [reply imports/plugins/custom/payments-paystack/client/checkout/paystack.js:63]: Yes, it is. Thank you for the feedback. Noted.

Comment (Felglitzzz) [reply imports/plugins/custom/payments-paystack/client/checkout/paystack.js:63]: The reference is the same as or say works as the product id

File: .eslintrc (hunk 1 above)
Comment (Felglitzzz) [reply .eslintrc:30]: No it isn't the reason

**Pull request merged**